go/engine/login_provision.go

```diff
@@ -715,17 +718,30 @@ func (e *loginProvision) chooseDevice(m libkb.MetaContext, pgp bool) (err error)
 	if len(id) == 0 {
 		// they chose not to use a device
 		m.Debug("user has devices, but chose not to use any of them")
-		if pgp {
-			if hasPUK {
-				m.Debug("user has a per-user-key, not attempting pgp provision")
-				return libkb.ProvisionViaDeviceRequiredError{}
+		if pgp && !hasPUK {
+			// they have pgp keys, so try that:
+			if err := e.tryPGP(m); err != nil {
+				// TODO: consider this flow with autoreset
+				return err
 			}
+		}
 
-			// they have pgp keys, so try that:
-			return e.tryPGP(m)
+		m.G().Log.Info(`
+The only way to provision this device is with access to one of your existing
+devices. You can try again later, or if you have lost access to all your
+existing devices you can reset your account and start fresh:`)
```

Comment:
I think you should be using the UI to communicate with the user, otherwise things like this won't show up in the GUI.  This should be in ProvisionUI.

Reply:
It's a CLI-only thing, this text was previously displayed by CmdLogin.errProvisionUnavailable() - I kept it this way for now because we don't have any design stuff just yet. It will be functional as-is, we just won't have this explanation (we'll just see "Please enter the Keybase password for %s (%d+ characters) if you know it"). I don't know if it's worth changing it just yet.

Reply:
... and the more important aspect of the "lack of feedback in the GUI" is the sole fact that we don't show anything there _after_ you join the ARP. I could probably display it using the progress GUI call?

Since you're here, a somewhat relevant question: how will all of this work if the GUI doesn't implement the prompt methods in LoginUI? Will the calls error? I wanted to ask that stuff last week but we couldn't even get the actual reseting running.

Reply:
They will be forced to handle any new functions added to a UI interface.  I believe `flow` will block it until they do.  When that happens, just reach out to @keybase/react-hackers and someone there will fix it up on their side.

Reply:
until the GUI implements the flow and we want to enable autoresets on prod we should hide the new functionality behind feature flags

Reply:
@pzduniak if you don't move this over to provisionUI you should add a TODO with a ticket where it will be done